File: DockerLogDriver/src/main/java/de/derteufelqwe/driver/endpoints/VolumeDriverCapabilitiesEP.java
package de.derteufelqwe.driver.endpoints;

import de.derteufelqwe.driver.messages.VolumeDriver;

import java.io.Serializable;

public class VolumeDriverCapabilitiesEP extends Endpoint<VolumeDriver.RCapabilities, VolumeDriver.Capabilities> {

    public VolumeDriverCapabilitiesEP(String data) {
        super(data);
    }

    @Override
    protected VolumeDriver.Capabilities process(VolumeDriver.RCapabilities request) {
        return new VolumeDriver.Capabilities();
    }

    @Override
    protected Class<? extends Serializable> getRequestType() {
        return VolumeDriver.RCapabilities.class;
    }

    @Override
    protected Class<? extends Serializable> getResponseType() {
        return VolumeDriver.Capabilities.class;
    }
}

File: DockerLogDriver/src/main/java/de/derteufelqwe/driver/endpoints/VolumeDriverCapabilitiesEP.kt
package de.derteufelqwe.driver.endpoints

import de.derteufelqwe.driver.messages.VolumeDriver
import java.io.Serializable

class VolumeDriverCapabilitiesEP(data: String?) : Endpoint<VolumeDriver.RCapabilities, VolumeDriver.Capabilities>(data) {

    override fun process(request: VolumeDriver.RCapabilities): VolumeDriver.Capabilities {
        return VolumeDriver.Capabilities()
    }

    override fun getRequestType(): Class<out Serializable?> {
        return VolumeDriver.RCapabilities::class.java
    }

    override fun getResponseType(): Class<out Serializable?> {
        return VolumeDriver.Capabilities::class.java
    }
}